modules/sliding-sync/default.nix

```diff
@@ -0,0 +1,123 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.matrix-next.sliding-sync;
+  inherit (lib)
+    mkEnableOption
+    mkOption
+    types
+    mkIf;
+in
+{
+  options.services.matrix-next.sliding-sync = {
```

Comment:
`options.services.matrix-next.sliding-sync`

right now the synapse module is still `options.services.matrix-synapse-next`

We should probably move this to `options.services.matrix-next.synapse` here or in a separate PR